apps/shopifyorders/shopify_orders.star

```diff
@@ -0,0 +1,213 @@
+        label = "orders last 365 days"
+    else:
+        label = "orders, {} - {}".format(start_date, end_date)
+
+    return render.Root(
+        child = render.Stack(
+            children = [
+                render.Image(src = base64.decode(IMAGE_PICTURE_FRAME_BG)),
+                render.Box(
+                    padding = 5,
+                    child = render.Column(
+                        cross_align = "center",
+                        children = [
+                            render.WrappedText(
+                                align = "center",
+                                content = value,
+                                color = COLOR_ALOE,
+                                font = FONT_TOM_THUMB,
+                            ),
+                            render.WrappedText(
+                                align = "center",
+                                content = label,
+                                font = FONT_TOM_THUMB,
+                            ),
+                        ],
+                    ),
+                ),
+            ],
+        ),
+    )
+
+date_range_options = [
+    schema.Option(
+        display = "Past 24 hours",
+        value = "last_day",
+    ),
+    schema.Option(
+        display = "Past 7 days",
+        value = "last_7_days",
+    ),
+    schema.Option(
+        display = "Past 30 days",
+        value = "last_30_days",
+    ),
+    schema.Option(
+        display = "Past 365 days",
+        value = "last_365_days",
+    ),
+    schema.Option(
+        display = "Custom date range",
+        value = "custom",
+    ),
+]
+
+def date_range_custom_options(relativeDate):
+    if relativeDate == "custom":
+        return [
+            schema.DateTime(
+                id = "startDate",
+                name = "Start Date",
+                desc = "Start date for the orders data",
+                icon = "gear",
+            ),
+            schema.DateTime(
+                id = "endDate",
+                name = "End Date",
+                desc = "End date for the orders data",
+                icon = "gear",
+            ),
+        ]
+    else:
+        return []
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "counterId",
+                name = "Counter ID",
+                desc = "Unique ID of the counter set up in the Counter app for Shopify",
+                icon = "shopify",
+            ),
+            schema.Dropdown(
+                id = "relativeDate",
+                name = "Date",
+                desc = "The date range for the orders data",
+                icon = "gear",
+                default = date_range_options[0].value,
+                options = date_range_options,
+            ),
+            schema.Generated(
+                id = "generated",
+                source = "relativeDate",
+                handler = date_range_custom_options,
```

Comment:
I noticed the generated date pickers don't work in the dev tools, but I think it's a bug in pixlet :sob: Will look to confirm once this is in our backend.